### PR TITLE
[aricpp] new port

### PR DESCRIPTION
**Pull request by daniele77**
This PR refers to Issue #17162

**Describe the pull request**

- #### What does your PR fix?  
     Add the new library `aricpp`

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
    all , No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  `Yes`

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/aricpp/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    SHA512 b1e02b9ba9afc1f3315e238cd61b98a8d28eee08ddbaccaf171aa77d27ecec2b3abfaa5aae6905f9c2a1c83b0095a135f2186c977a0ae0cfafb48e3690814183 
+    HEAD_REF master
+)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
+)
+
+vcpkg_install_cmake()
```
Comment (JackBoosY) [inline ports/aricpp/portfile.cmake:14]: These 2 functions are deprecated, please use `vcpkg_cmake_configure` and `vcpkg_cmake_install` then add dependency `vcpkg-cmake`.

File: ports/aricpp/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/aricpp)
```
Comment (JackBoosY) [inline ports/aricpp/portfile.cmake:15]: Please also switch it to `vcpkg_cmake_config_fixup` and add dependency `vcpkg-cmake-config`.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/aricpp/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    REF v1.1.1
+    SHA512 b1e02b9ba9afc1f3315e238cd61b98a8d28eee08ddbaccaf171aa77d27ecec2b3abfaa5aae6905f9c2a1c83b0095a135f2186c977a0ae0cfafb48e3690814183 
+    HEAD_REF master
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH ${SOURCE_PATH}
+)
+
+vcpkg_cmake_install()
```
Comment (JackBoosY) [inline ports/aricpp/portfile.cmake:13]:
```suggestion
vcpkg_cmake_install()

vcpkg_copy_pdbs()
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Review by JackBoosY — CHANGES_REQUESTED** [review]
```
CMake Error at D:/downloads/tools/cmake-3.20.2-windows/cmake-3.20.2-windows-i386/share/cmake-3.20/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Boost (missing: Boost_INCLUDE_DIR system) (Required is at
  least version "1.66")
```
Seems missing some boost components in dependency list.

File: ports/aricpp/vcpkg.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "aricpp",
+  "version-semver": "1.1.1",
+  "port-version": 1,
+  "description": "Asterisk ARI interface bindings for modern C++",
+  "homepage": "https://github.com/daniele77/aricpp",
+  "dependencies": [
+    {
```
Comment (JackBoosY) [inline ports/aricpp/vcpkg.json:8]:
```suggestion
    "boost-asio",
    "boost-beast",
    "boost-system",
    {
```

File: ports/aricpp/vcpkg.json [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
+    },
+    "boost-system",
+    "boost-asio",
+    "boost-beast"
```
Comment (JackBoosY) [inline ports/aricpp/vcpkg.json:18]:
```suggestion
    }
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: versions/a-/aricpp.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
+{
+  "versions": [
+    {
+      "git-tree": "4bbfa3f90947fa234f7d3e3ab43ca80c5007148c",
+      "version-string": "1.1.1",
```
Comment (JackBoosY) [inline versions/a-/aricpp.json:5]:
```suggestion
      "version-semver": "1.1.1",
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by JackBoosY** [comment]
Waiting for merge https://github.com/microsoft/vcpkg/pull/18116.

**Review by JackBoosY — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Awesome, thanks for the great library and PR!

**Pull request merged**